lib/models/addon.js

```diff
@@ -1206,31 +1205,21 @@ let addonProto = {
       });
       let lintTemplateTrees = this._eachProjectAddonInvoke('lintTree', ['templates', addonTemplates]);
 
-      trees = trees.concat(lintAddonJsTrees, lintTemplateTrees);
-    }
+      lintTrees = lintTrees.concat(
+        lintAddonJsTrees,
+        lintTemplateTrees
+      );
 
-    let addonTestSupportPath = this._treePathFor('addon-test-support');
-    if (fs.existsSync(addonTestSupportPath)) {
-      let addonTestSupportTree = new Funnel(addonTestSupportPath, { destDir: 'addon-test-support' });
-      let lintAddonTestSupportJsTrees = this._eachProjectAddonInvoke('lintTree', ['addon-test-support', addonTestSupportTree]);
-      trees = trees.concat(lintAddonTestSupportJsTrees);
     }
 
-    let appPath = this._treePathFor('app');
-    if (fs.existsSync(appPath)) {
-      let appTree = new Funnel(appPath, { destDir: 'app' });
-      let lintAppJsTrees = this._eachProjectAddonInvoke('lintTree', ['app', appTree]);
-      trees = trees.concat(lintAppJsTrees);
-    }
+    lintTrees = lintTrees.concat(
+      this._lintTree('test-support'),
+      this._lintTree('app'),
+      this._lintTree('addon-test-support')
+    ).filter(Boolean);
+
 
-    let testSupportPath = this._treePathFor('test-support');
-    if (fs.existsSync(testSupportPath)) {
-      let testSupportTree = new Funnel(testSupportPath, { destDir: 'test-support' });
-      let lintTestSupportJsTrees = this._eachProjectAddonInvoke('lintTree', ['test-support', testSupportTree]);
-      trees = trees.concat(lintTestSupportJsTrees);
```

Comment:
note that this was `concat()`, not `push()`. this might have to be flattened before we can `concat()` the `_lintTree()` results. 🤔

Reply:
How does this look?